### PR TITLE
silversearcher-ag: Add package

### PR DESCRIPTION
**Pull request by nikil**
Maintainer: me
Compile tested: ar71xx, x86_64
Run tested: x86_64

Description: very fast grep-like program, altenative to ack-grep

Signed-off-by: Nikil Mehta nikil.mehta@gmail.com

### CODE REVIEW

File: utils/silversearcher-ag/Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+define Package/silversearcher-ag/description
+	The Silver Searcher is grep-like program implemented by C.
+	An attempt to make something better than ack-grep.
+	.
+	It searches pattern about 3–5x faster than ack-grep.
+	It ignores file patterns from your .gitignore and .hgignore.
+endef
+
+define Package/collectd/conffiles
+/etc/bash_completion.d/ag
```
Comment (diizzyy) [inline utils/silversearcher-ag/Makefile]: Change this or provide an option to bash stuff?

File: utils/silversearcher-ag/Makefile [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=silversearcher-ag
+PKG_VERSION:=0.33.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=the_silver_searcher-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=http://geoff.greer.fm/ag/releases/
+PKG_MD5SUM:=27aa17dbf4e0324defa341764852ba58
```
Comment (diizzyy) [inline utils/silversearcher-ag/Makefile]: Please use SHA256

File: utils/silversearcher-ag/Makefile [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#
+# Copyright (C) 2006-2016 OpenWrt.org
```
Comment (diizzyy) [inline utils/silversearcher-ag/Makefile]: Please don't use OpenWrt.org copyright unless you are under contract.
https://www.lede-project.org/faq.html#who_holds_the_copyright_of_lede_code

File: utils/silversearcher-ag/Makefile [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+endef
+
+define Build/Prepare
+	tar xvfz $(DL_DIR)/$(PKG_SOURCE) -C $(PKG_BUILD_DIR) --strip-components=1
+endef
+
+define Package/silversearcher-ag/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/ag $(1)/usr/bin/
+	$(INSTALL_DIR) $(1)/etc/bash_completion.d
```
Comment (diizzyy) [inline utils/silversearcher-ag/Makefile]: Same as above

File: utils/silversearcher-ag/Makefile (hunk 1 above)
Comment (zorun) [inline utils/silversearcher-ag/Makefile]: Also, there is a copy-paste typo (collectd instead of silversearcher-ag)

Comment (nikil) [inline utils/silversearcher-ag/Makefile]: I've noticed that no other packages add bash completion, so I'll just remove this.  It's not necessary.

**Comment by nikil** [comment]
Done - thanks for the comments.

**Comment by yousong** [comment]
ag is good, but do you really need this tool on OpenWrt/LEDE?  Is grep really that bad for your workflow?  I am just wondering the use case.

**Comment by nikil** [comment]
In my NAS environment, ag makes a huge difference (> 10x speedup) for some of the tools I'm writing.

The reason I ported this to OpenWrt was because I find it useful.  I suppose others can judge for themselves?

**Comment by yousong** [comment]
@nikil I understand the fact that ag is good at what it is created for.  I used it everyday to search through large chunks of text files: source code repositories, or notes written in markdown.

ag is certainly a useful tool, but I think the need for searching through a lot of text files regularly is not supposed to be a common task on OpenWrt/LEDE.  Providing a description of current or possible future use cases will facilitate the inclusion of new packages with the fact that "All the apps get compiled for all ~50 targets by buildbots, so adding new apps en masse could slow down buildbot compilation." (https://github.com/openwrt/packages/issues/3579#issuecomment-263139423)

**Comment by nikil** [comment]
Oh, I didn't know that OpenWrt was having a problem with buildbot throughput.

Sure, I can document my use case.  I'm on a research project that is developing some modular intrusion detection software that's intended to be run on routers and NAS appliances.  There's a script interface for writing plugins, and it make a big difference for performance to have something that can search rules (files) quickly.

**Comment by neheb** [comment]
@nikil if you want this in, please rebase to current master.

**Comment by diizzyy** [comment]
Closing due to timeout and targeting a very small userbase

**Pull request closed without merging**